### DISCUSSION TITLE
How does the regex work?

### DISCUSSION THREAD

**Discussion by shiny-ribbon** [proginosko/LeechBlockNG#439] [Q&A]
I thought I knew a little bit of regex, but guess not.
I couldn't figure out how to make a regex to match a site homepage but not its subpages
I tried some variations of

`https:\/\/reddit\.com`
`^https:\/\/www\.reddit\.com$`
`^(https:\/+(www\.)?reddit\.com)$`

No luck

**Comment by shiny-ribbon** [comment]
Never mind, ended up finding a tapatalk thread where the person had the exact same problem. In the end I went from there.
My final regex looks like this

`^(https?|file):\/+((www\.)?(reddit|discord)\.com(\/(r|message)\/(all|popular|inbox|portugal))?\/?)$`

It covers the reddit homepage, a few specific subpages, but only those, as well as discord for good measure
As for interpreting it? I can't do it

**Comment by proginosko** [comment]
Yes, it's tricky, but that should do it.
